File: medium/3689-maximum-total-subarray-value-i/solution.java
/**
 * 3689. Maximum Total Subarray Value I
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public long maxTotalValue(int[] nums, int k) {
        int globalMax = nums[0];
        int globalMin = nums[0];

        for (int num : nums) {
            globalMax = Math.max(globalMax, num);
            globalMin = Math.min(globalMin, num);
        }

        return (long) k * (globalMax - globalMin);
    }
}

File: medium/3689-maximum-total-subarray-value-i/solution.py
from typing import List


class Solution:
    def maxTotalValue(self, nums: List[int], k: int) -> int:
        """
        3689. Maximum Total Subarray Value I
        Time: O(n)
        Space: O(1)
        """
        return k * (max(nums) - min(nums))
